### ISSUE TITLE
SST Compiler warnings on Mac / Clang

### ISSUE THREAD

**Issue by JasonRuonanWang** [ornladios/ADIOS2#552]
[ 43%] Building CXX object source/adios2/CMakeFiles/adios2.dir/operator/compress/CompressBZip2.cpp.o
/Users/w4g/dropbox/adios2_sst/source/adios2/engine/sst/SstReader.cpp:48:29: warning: ISO C++ specifies that qualified reference to 'SstReader' is a
      constructor name rather than a type in this context, despite preceding 'typename' keyword [-Winjected-class-name]
        typename SstReader::SstReader *Reader =
                            ^
/Users/w4g/dropbox/adios2_sst/source/adios2/engine/sst/SstReader.cpp:49:50: warning: ISO C++ specifies that qualified reference to 'SstReader' is a
      constructor name rather than a type in this context, despite preceding 'typename' keyword [-Winjected-class-name]
            reinterpret_cast<typename SstReader::SstReader *>(reader);
                                                 ^
/Users/w4g/dropbox/adios2_sst/source/adios2/engine/sst/SstReader.cpp:77:29: warning: ISO C++ specifies that qualified reference to 'SstReader' is a
      constructor name rather than a type in this context, despite preceding 'typename' keyword [-Winjected-class-name]
        typename SstReader::SstReader *Reader =
                            ^
/Users/w4g/dropbox/adios2_sst/source/adios2/engine/sst/SstReader.cpp:78:50: warning: ISO C++ specifies that qualified reference to 'SstReader' is a
      constructor name rather than a type in this context, despite preceding 'typename' keyword [-Winjected-class-name]
            reinterpret_cast<typename SstReader::SstReader *>(reader);
                                                 ^

**Comment by eisenhauer** [comment]
Yeah, a clang update a while ago started not liking this.  I have a fix, but it's part of the dyn memory WIP, so this should be addressed when that gets finalized.
